src/Compilers/CSharp/Portable/Symbols/Compilation_WellKnownMembers.cs

```diff
@@ -141,6 +141,38 @@ internal bool IsExceptionType(TypeSymbol type, ref HashSet<DiagnosticInfo> useSi
             return IsEqualOrDerivedFromWellKnownClass(type, WellKnownType.System_Exception, ref useSiteDiagnostics);
         }
 
+        internal bool IsTupleUnderlyingType(TypeSymbol typeToCheck, int requiredArity)
```

Comment:
Consider moving to `TupleTypeSymbol` as static helper?
What about calling this `IsWellKnownTupleType`? Most likely we will keep it that way (and not allow ambiguous situations).